src/nvim/api/buffer.c

```diff
@@ -1091,25 +1063,9 @@ void nvim_buf_delete(Buffer buffer, Dictionary opts, Error *err)
     return;
   }
 
-  bool force = false;
-  bool unload = false;
-  for (size_t i = 0; i < opts.size; i++) {
-    String k = opts.items[i].key;
-    Object v = opts.items[i].value;
-    if (strequal("force", k.data)) {
-      force = api_object_to_bool(v, "force", false, err);
-    } else if (strequal("unload", k.data)) {
-      unload = api_object_to_bool(v, "unload", false, err);
-    } else {
-      VALIDATE_S(false, "'opts' key", k.data, {
-        return;
-      });
-    }
-  }
+  bool force = HAS_KEY(opts, buf_delete, force) ? opts->force : false;
 
-  if (ERROR_SET(err)) {
-    return;
-  }
+  bool unload = HAS_KEY(opts, buf_delete, unload) ? opts->unload : false;
```

Comment:
```suggestion
  bool unload = opts->unload;
```

Reply:
Fixed in af08794ed29ae67c508d54e9d97b5341daec2892.